mlflow/pyfunc/backend.py

```diff
@@ -127,9 +127,12 @@ def _execute_in_conda_env(conda_env_path, command, install_mlflow, command_env=N
         command_env = os.environ
     env_id = os.environ.get("MLFLOW_HOME", VERSION) if install_mlflow else None
     conda_env_name = _get_or_create_conda_env(conda_env_path, env_id=env_id)
-    activate_path = _get_conda_bin_executable("activate")
-    activate_conda_env = ["source {0} {1} 1>&2".format(activate_path, conda_env_name)]
-
+    conda_path = _get_conda_bin_executable("conda")
+    activate_conda_env = []
+    if conda_path != 'conda':
+        activate_conda_env = ['source ' + os.path.dirname(conda_path) +
```

Comment:
Apply this change to `mlflow.project._get_conda_command`. This way even `mlflow.project` will be patch.

Reply:
Updated the mlflow.project._get_conda_command

Reply:
Thanks for this fix, I think unfortunately this logic will break with older condas e.g. conda 4.3 (i.e. for older condas, we'll fall through to running `conda activate` below, which doesn't exist). It'd also be nice to avoid the indirection of checking whether `conda_path != 'conda'` and just directly check if conda environment variables are set - thus I'd propose the following changes:


* If CONDA_EXE or MLFLOW_CONDA_HOME is set, assume that we're in a new version of conda & add `'source ' + os.path.dirname(conda_path) + '/../etc/profile.d/conda.sh'` to `activate_conda_env
* Otherwise, default to running `source activate`

Reply:
I am not sure to get why we are not using `_get_conda_command`. It will do the same and add some extra support for windows. Fix fox `CONDA_EXE` or `MLFLOW_CONDA_HOME` has been applied to `_get_conda_command`.

Reply:
@nlaille Yeah your right.We are in a way doing the same stuff in both the places.Rather than duplicating the same code we better use the `_get_conda_command` method itself.Should I make the change?

Reply:
It would make more sense.

Reply:
Refactored the same